src/schemas.ts

```diff
@@ -0,0 +1,58 @@
+import * as yup from "yup";
+import * as ethers from "ethers";
+
+const testIsAddress = {
+  name: "is-address",
+  message: "${path} is not an address",
+  test: isAddress
+};
+
+const testIsBigNumberish = {
+  name: "is-big-numberish",
+  message: "${path} is not BigNumberish",
+  test: isBigNumberish
+};
+
+const recipientSchema = yup.object({
+  amount: yup.string().test(testIsBigNumberish).defined(),
+  account: yup.string().test(testIsAddress).defined(),
+  metadata: yup
+    .object({
+      amountBreakdown: yup
+        .object({
+          welcomeTravelerRewards: yup
+            .string()
+            .test(testIsBigNumberish)
```

Comment:
Should we add checks that these are all non negative?

Reply:
I suggest to add a check to make sure numbers are not in scientific notation (e+)